### DISCUSSION TITLE
Uncertainties for data/prediction ratio plots

### DISCUSSION THREAD

**Discussion by alexander-held** [scikit-hep/hist#196] [Q&A]
The `.plot_ratio` API introduces a way to plot ratios of histograms, including uncertainties for this ratio. It is common to show uncertainties for (a stack of histograms or a single histogram with) model predictions and data statistical uncertainties separately in ratio plots. A random example is [this plot](https://atlas.web.cern.ch/Atlas/GROUPS/PHYSICS/CONFNOTES/ATLAS-CONF-2021-014/fig_03a.pdf).

Is it possible to create a plot like this with `hist`? Possibly related: #192, and #117 in case something like this fits better into `mplhep`.

cc @matthewfeickert

**Comment by matthewfeickert** [comment]
As was brought up in a separate conversation for my own clarity with @alexander-held, @lukasheinrich, and @kratsg, the uncertainties shown for the relative model uncertainty (the ratio of the model predictions with full uncertainties to the model predictions themselves) should be added, and I missed these previously when adding the uncertainties between the ratio of the data and the model. (I guess more generally this would be the relative uncertainties on the denominator histogram and then the uncertainties of the ratio between the numerator and denominator histograms).

This is [already done in `cabinetry.visualize.data_MC`](https://github.com/alexander-held/cabinetry/blob/5b1684fb221d9d8ceec0229b3834e5e6025b8347/src/cabinetry/contrib/matplotlib_visualize.py#L126-L137)

```python
    # add uncertainty band around y=1
    rel_mc_unc = total_model_unc / total_yield
    ax2.bar(
        bin_centers,
        2 * rel_mc_unc,
        width=bin_width,
        bottom=1 - rel_mc_unc,
        fill=False,
        linewidth=0,
        edgecolor="gray",
        hatch=3 * "/",
    )
```

as well as in [`coffea.hist.plotratio`](https://github.com/CoffeaTeam/coffea/blob/ee3e76d01df45b1453f91ac9ccc00a67074f14ad/coffea/hist/plot.py#L450-L460)

```python
        if denom_fill_opts is not None:
            unity = numpy.ones_like(sumw_denom)
            denom_unc = poisson_interval(unity, sumw2_denom / sumw_denom ** 2)
            opts = {"step": "post", "facecolor": (0, 0, 0, 0.3), "linewidth": 0}
            opts.update(denom_fill_opts)
            ax.fill_between(
                edges,
                numpy.r_[denom_unc[0], denom_unc[0, -1]],
                numpy.r_[denom_unc[1], denom_unc[1, -1]],
                **opts
            )
```

**Examples:**

`cabinetry`:

![Signal_region_postfit](https://user-images.githubusercontent.com/5142394/114952253-82c22800-9e1b-11eb-8e28-342999490173.png) 

`coffea`:
[![coffea ratio plot](https://coffeateam.github.io/coffea/_images/notebooks_histograms_46_0.png)](https://coffeateam.github.io/coffea/notebooks/histograms.html#Styling)

I'll also tag @nsmith- to see if there's any thoughts on details of implementation to think about adding that I'm missing.
